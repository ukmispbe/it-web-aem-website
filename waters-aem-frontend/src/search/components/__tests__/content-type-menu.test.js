import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import ContentTypeMenu from '../content-type-menu';

describe('Feature: ContentTypeMenu Component', () => {
    const props = {
        heading: '',
        items: [],
        onClick: jest.fn()
    };

    const propsOneItem = {
        ...props,
        items: [
            {
                count: 5,
                facetName: "applicationnote_facet",
                facetTranslation: "Application Note",
                facetValue: "Application Note",
                orderedFacets: []
            }
        ]
    };

    describe('Scenario: Rendering', () => {
        describe('When no content types are specified', () => {
            it('Then it should match snapshot', () => {
                const json = renderer.create(<ContentTypeMenu {...props} />);

                expect(json).toMatchSnapshot();
            });
        });

        describe('When content types are specified', () => {
            it('Then it should match snapshot', () => {
                const json = renderer.create(<ContentTypeMenu {...propsOneItem} />);
                
                expect(json).toMatchSnapshot();

            });
        });
    });

    describe('Scenario: User Interaction', () => {
        describe('When the content type is clicked', () => {
            it('Then is should call the click handler property', () => {
                const spyOnClick = spyOn(propsOneItem, 'onClick');
                const wrapper = shallow(<ContentTypeMenu {...propsOneItem} />);
                const contentType = wrapper.find('.content-type-menu-container__item');
                
                contentType.simulate('click');

                expect(spyOnClick).toHaveBeenCalled();

                spyOnClick.mockRestore();
            });
        });
    });
});