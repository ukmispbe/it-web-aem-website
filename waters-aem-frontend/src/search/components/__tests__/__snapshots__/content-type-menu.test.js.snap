// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: ContentTypeMenu Component Scenario: Rendering When content types are specified Then it should match snapshot 1`] = `
<div
  className="content-type-menu-container"
>
  <div
    className="content-type-menu-container__heading"
  >
    Filter By
  </div>
  <div
    className="content-type-menu-container__body"
  >
    <div
      className="content-type-menu-container__item"
      onClick={[Function]}
    >
      <div>
        <a
          data-count=" (5)"
          href="javascript:{0}"
        >
          Application Note
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Feature: ContentTypeMenu Component Scenario: Rendering When no content types are specified Then it should match snapshot 1`] = `
<div
  className="content-type-menu-container"
>
  <div
    className="content-type-menu-container__heading"
  >
    Filter By
  </div>
  <div
    className="content-type-menu-container__body"
  />
</div>
`;
